src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,107 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.northbound.dto.v2.haflows.HaFlow
+
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""Verify that history records are created for the basic actions applied to Ha-Flow.""")
+class HaFlowHistorySpec extends HealthCheckSpecification {
+
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Shared
+    boolean isDeleted
+
+    @Tidy
+    def "User can change a Ha-Flow and get its history event - #type"() {
+        given: "HA-FLow"
+        def swT = topologyHelper.switchTriplets[0]
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        when: "#description the flow"
+        change(haFlow)
+
+        then: "Correct event appears in ha flow history"
+        haFlowHelper.getHistory(haFlow.haFlowId).getEntryByType(type).size() == 1
+
+        cleanup:
+        !isDeleted && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+
+        where:
+        type      | change
+
+//    @Ignore("https://github.com/telstra/open-kilda/issues/5320")
```

Comment:
Spock has mechanism to ignore specific parametrized tests (@IgnoreIf). Please, use the one, if possible.

Reply:
spock has no ability to ignore iteration based on data, it uses system variables like os, params etc for the such ignorance
the same for the @requires